addons/udes_stock_routing/models/procurement_group.py

```diff
@@ -0,0 +1,11 @@
+from odoo import models, api
+
+class ProcurementGroup(models.Model):
+    _inherit = "procurement.group"
+
+    @api.model
+    def _get_rule_domain(self, location, values):
+        """Extend rule domain search to exclude rules which run on stock assignment"""
+        res = super()._get_rule_domain(location, values)
+        res.append(("u_run_on_assign", "=", False))
```

Comment:
I think is safer like this
```suggestion
        res = expression.AND([res, [("u_run_on_assign", "=", False)]])
```

Reply:
bece12538bfc1af897a0c3b02329ab72292123e8